internal/logstore/debouncer.go

```diff
@@ -0,0 +1,87 @@
+package logstore
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	// Using this time.Time replacement enables us to control timers and tickers in the unit tests
+	"github.com/benbjohnson/clock"
+	"github.com/zitadel/logging"
+)
+
+type bulkSink interface {
+	sendBulk(ctx context.Context, bulk []LogRecord) error
+}
+
+var _ bulkSink = bulkSinkFunc(nil)
+
+type bulkSinkFunc func(ctx context.Context, items []LogRecord) error
+
+func (s bulkSinkFunc) sendBulk(ctx context.Context, items []LogRecord) error {
+	return s(ctx, items)
+}
+
+type debouncer struct {
+	ctx      context.Context
```

Comment:
Will have to think about that

Reply:
I understand that it i bad practice if you use the context for transactions, e.g. functions that are externally called. However, the debouncer triggers transactions itself on certain criteria. i.e. it defines itself when the bulk has to be shipped.

As shipping involves side effects (db access), a context is needed. The intent here is, that we shouldn't just create new background contexts but rather use the apps background contexts, as this can be cancelled by os signals.

I renamed the variable and commented it, to make it more explicit. Do you agree, or do you have a better solution?

Reply:
discussed